hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -206,15 +206,7 @@ private List<String> getPartitionPathsForIncrementalCleaning(HoodieCleanMetadata
    */
   private List<String> getPartitionPathsForFullCleaning() {
     // Go to brute force mode of scanning all partitions
-    try {
-      // Because the partition of BaseTableMetadata has been deleted,
-      // all partition information can only be obtained from FileSystemBackedTableMetadata.
```

Comment:
trying to understand this previous comment, as questioned here https://github.com/apache/hudi/pull/5169/files#r957849431

under what scenario has the metadata table's partition being delete? even deleting synchronously, we should make use of metadata table first then delete metadata and data altogether?

Reply:
yeah. when Sagar wrote that code, here is how DELETE_PARTITION was implemented. 
it wasn't well thought out design/impl. The way it was designed was, partition was synchronously deleted in data table and replace commit was relayed to MDT where in all file groups will be marked as invalid. but the partition in MDT remained. 

if you remember, I chimed in and rewrote the entire impl, where in we made DELETE_PARTITION lazy.

Reply:
guess we can't go w/ this change. essentially what we are doing in cleaning is

1. fetch list of partitions to clean
2. Fetch list of files to clean
3. for those partitions where there is no valid file groups, we mark the partition to be deleted and physically delete the directory (lazy execution of DELETE_PARTITION)

So, in step1, we have to go w/ FS based listing. if we revert to metadata table based listing, we may not get the deleted partition only. and so the physical directory will never get deleted. 

Need to think about an optimized way to go about this. for now, will punt on the patch.

Reply:
One option is.
during DELETE_PARTITION, we will not delete it from metadata table. 
When cleaner comes around and detects there are not valid file groups, it will physically delete it and will relay it to metadata table. at this juncture, metadata table will remove the partition of interest.

Reply:
@nsivabalan I think your suggestion makes sense given that we now do lazy cleaning. If `files` partition of metadata table exists then, most likely, the record is there with `isDeleted` set to true. The metadata based fs view should understand the [isDeleted](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L50) flag of `HoodieMetadataFileInfo`.